### ISSUE TITLE
Flag throwing non Error/Exception types

### ISSUE THREAD

**Issue by kevmoo** [dart-lang/linter#239]
```dart
throw 'some string';
```

or `null`, number, etc.

**Comment by alexeieleusis** [comment]
What about a related rule to not to catch `Error`? Like not catching `Throwable` in java?

**Comment by bwilkerson** [comment]
Not sure how useful that would be (sometimes it's useful to catch such exceptions in order to log them, for example), but the nice thing about lints is that they're opt-in, so it won't hurt to have it. The rule would probably need to be that you shouldn't catch one without re-throwing it. Not sure what to do with code that catches *anything*:

```
try {} catch (e) {}
```

**Comment by alexeieleusis** [comment]
Agreed, either rethrowing or finishing the program. Will add that to my TODO list.
